### PR TITLE
Add field_access to commerce_cart module

### PR DESCRIPTION
**Pull request by jaredsmith**
Here's my initial stab at a pull request for this issue.  

I'm not familiar with the Field API, so I'm mostly just copying/pasting from the 1.x code, so feel free to point out anything that might have changed in for d8.

### CODE REVIEW

File: modules/cart/commerce_cart.module [hunk 1]
```diff
@@ -122,6 +122,44 @@ function commerce_cart_form_commerce_order_type_form_alter(array &$form, FormSta
 }
 
 /**
+ * Implements hook_form_FORM_ID_alter() for 'field_ui_field_edit_form'.
+ */
+function commerce_order_form_field_config_edit_form_alter(array &$form, FormStateInterface $form_state) {
```
Comment (bojanz) [inline modules/cart/commerce_cart.module]: Wrong function name. Needs to be commerce_cart_form_field_config_edit_form_alter. The current version never gets called :)

File: modules/cart/commerce_cart.module [hunk 2]
```diff
@@ -122,6 +122,44 @@ function commerce_cart_form_commerce_order_type_form_alter(array &$form, FormSta
...
+  $field = $form_state->getFormObject()->getEntity();
+  if ($field->isLocked() === false && $field->get('entity_type') == 'commerce_line_item') {
```
Comment (bojanz) [inline modules/cart/commerce_cart.module]: we can do !$field->isLocked()

File: modules/cart/commerce_cart.module [hunk 3]
```diff
@@ -122,6 +122,44 @@ function commerce_cart_form_commerce_order_type_form_alter(array &$form, FormSta
...
+
+    // Get the current instance's line item form settings for use as default values.
+    $instance = $form['#instance'];
```
Comment (bojanz) [inline modules/cart/commerce_cart.module:132]: I doubt $form['#instance'] exists. Also, we don't prefix variables with commerce

Comment (jaredsmith) [inline modules/cart/commerce_cart.module:132]: OK, as I said -- I don't understand how this works in D8, so I can definitely use either some other examples to look at, or some guidance on how this works.

File: modules/cart/commerce_cart.module (hunk 1 above)
Comment (jaredsmith) [inline modules/cart/commerce_cart.module]: Fixed.

File: modules/cart/commerce_cart.module (hunk 2 above)
Comment (jaredsmith) [inline modules/cart/commerce_cart.module]: Fixed.

**Comment by bojanz** [comment]
We decided to implement this differently, by using a form mode instead.

**Pull request closed without merging**